### PR TITLE
Added sources to the ARP attributes form and connections API

### PR DESCRIPTION
**Pull request by MKodde**
The SP config's ARP form now enables the end user to select the source of the selected ARP Attribute.  Available sources are configured in the janus config file. The selected sources are stored in the database.

On a side note: in this feature branch, support for older PHP/HHVM versions are no longer verified. Travis test builds for PHP 5.6 remain.

### CODE REVIEW

File: src/Janus/ServiceRegistry/Bundle/CoreBundle/Service/ArpAttributeHelper.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    /**
+     * Merge the arp attributes and the sources.
+     *
+     * The array is stored like this:
+     *  [
+     *      attrName => [
+     *          [
+     *              'value' => 'filterValue'
+     *              'soure' => 'sourceName' (if specifically set)
+     *          ],     *          [
```
Comment (jorissteyn) [inline src/Janus/ServiceRegistry/Bundle/CoreBundle/Service/ArpAttributeHelper.php]: Minor formatting issue here.

File: src/Janus/ServiceRegistry/Bundle/CoreBundle/Service/ArpAttributeHelper.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     *              'value' => 'filterValue'
+     *          ],
+     *      ]
+     *  ]
+     * @param $attributes
+     * @param $sources
+     * @return array
+     */
+    public function mergeAttributes($attributes, $sources)
```
Comment (jorissteyn) [inline src/Janus/ServiceRegistry/Bundle/CoreBundle/Service/ArpAttributeHelper.php]: Why not type-hint here? Types are also missing from annotations.

File: src/Janus/ServiceRegistry/Entity/Connection.php [hunk 3]
```diff
@@ -146,7 +142,7 @@ public function __construct(
      * @param string|null $manipulationCode
      * @param bool $isActive
      * @param string|null $notes
-     * @param ConfigProxy $janusConfig
+     * @internal param ConfigProxy $janusConfig
```
Comment (jorissteyn) [inline src/Janus/ServiceRegistry/Entity/Connection.php]: @internal? Shouldn't we drop this line?

File: templates/editentity/arp.php [hunk 4]
```diff
@@ -31,14 +33,37 @@ function attributeNameValuePairId($arpAttrName, $value)
...
+                            </option>
+                            <?php endforeach; ?>
+                        </select>
+                    </td>
                     <td data-specify-values="<?php echo $arpSpecifyValues ? 'true' : 'false'; ?>">
                         <?php if ($arpAttributeUsed): ?>
-                            <?php foreach ($arp[$attribute['name']] as $value): ?>
+                            <?php
+                            foreach ($arp[$attribute['name']] as $value):
+                                $value = $arpAttributeHelper->getAttributeFilterValue($value);
```
Comment (jorissteyn) [inline templates/editentity/arp.php]: I understand you didn't introduce this problem, but this looks like XSS. Also, it's better to escape the value right where you print it. So, always do something like 'print escape($x)', never 'print $x'.

Comment (MKodde) [inline templates/editentity/arp.php]: This, and the other review notes have been taken care of.

File: app/config-dist/config_janus_core.yml [hunk 5]
```diff
@@ -113,39 +113,60 @@ janus_service_registry_core:
...
+          specify_values: true
+      'nlEduPersonStudyBranch (deprecated)':
+          name: 'urn:mace:surffederatie.nl:attribute-def:nlEduPersonStudyBranch'
+          specify_values: true
+      'nlStudielinkNummer (deprecated)':
+          name: 'urn:mace:surffederatie.nl:attribute-def:nlStudielinkNummer'
+      'nlDigitalAuthorIdentifier (deprecated)':
+          name: 'urn:mace:surffederatie.nl:attribute-def:nlDigitalAuthorIdentifier'
+      'collabPersonId (deprecated)':
+          name: 'urn:oid:1.3.6.1.4.1.1076.20.40.40.1'
```
Comment (thijskh) [inline app/config-dist/config_janus_core.yml]: Why have these definitions changed?

**Comment by thijskh** [comment]
Can we get a screenshot of the changed ARP tab in SR?

File: app/config-dist/config_janus_core.yml (hunk 5 above)
Comment (MKodde) [inline app/config-dist/config_janus_core.yml]: They where taken from the template `config_janus_core.yml` from the Deploy repo. Is this an example of boy scout rule gone bad?

**Comment by MKodde** [comment]
Here is a preview of the ARP Attributes form in JANUS

arp-form-with-sources

**Pull request merged**